perception/traffic_light_occlusion_predictor/src/occlusion_predictor.cpp

```diff
@@ -0,0 +1,245 @@
+// Copyright 2023-2026 the Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "traffic_light_occlusion_predictor/occlusion_predictor.hpp"
+
+namespace
+{
+
+traffic_light::Ray point2ray(const pcl::PointXYZ & pt)
+{
+  traffic_light::Ray ray;
+  ray.dist = std::sqrt(pt.x * pt.x + pt.y * pt.y + pt.z * pt.z);
+  ray.elevation = RAD2DEG(std::atan2(pt.y, std::hypot(pt.x, pt.z)));
+  ray.azimuth = RAD2DEG(std::atan2(pt.x, pt.z));
```

Comment:
[imo] In autoware.universe, it is recommended to use radians as the unit of angles unless there are special circumstances. Please use radians in other parts in this PR as well.

Reply:
I use degree in this package because when calculating the occlusion, a 2D grid table is built to represent the point cloud. The grid size of the table is one degree. If converting to radian at first, we still need to convert back to degree when building the table which could be very confusing.